controller/src/main/java/io/pravega/controller/store/stream/AbstractStreamMetadataStore.java

```diff
@@ -646,19 +637,54 @@ private Scope getScope(final String scopeName) {
 
     private CompletableFuture<Long> findNumMerges(String scopeName, String streamName, Executor executor) {
         return getScaleMetadata(scopeName, streamName, null, executor).thenApply(scaleMetadataList -> {
-            long size = scaleMetadataList.size();
+            int size = scaleMetadataList.size();
             long totalNumMerges = 0;
-
-            for (int i = 0, j = 1; j < size; j++) {
-                long countMergesEventI = scaleMetadataList.get(i).getSegments().size();
-                long countMergesEventJ = scaleMetadataList.get(j).getSegments().size();
-                totalNumMerges = (countMergesEventI > countMergesEventJ) ? (countMergesEventI - countMergesEventJ) : 0;
+            List<Segment> segmentList1, segmentList2;
+            boolean isDescendingOrder = isDescendingOrder(scaleMetadataList);
+
+            for (int i = 0; i < size - 1; i++) {
+                segmentList1 = scaleMetadataList.get(i).getSegments();
+                segmentList2 = scaleMetadataList.get(i+1).getSegments();
+                if (isDescendingOrder) {
+                    totalNumMerges += findSegmentSplitsMerges(segmentList1, segmentList2);
+                } else {
+                    totalNumMerges += findSegmentSplitsMerges(segmentList2, segmentList1);
+                }
             }
 
             return totalNumMerges;
         });
     }
 
+    private boolean isDescendingOrder(List<ScaleMetadata> scaleMetadataList) {
+        int size = scaleMetadataList.size();
+        boolean isDescendingOrder = true;
+        if (size > 1) {
+            if (scaleMetadataList.get(0).getTimestamp() < scaleMetadataList.get(1).getTimestamp()) {
```

Comment:
you can simply do 
isDescendingOrder = scaleMetadataList.get(0).getTimestamp() < scaleMetadataList.get(1).getTimestamp()